### PR TITLE
 Zoom class added

### PR DESCRIPTION
**Pull request by tanvidadu**
#1044 Class to add zoom functionality added.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/Zoom.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import android.graphics.PointF;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.View.OnTouchListener;
+import android.widget.ImageView;
+
+public class Zoom  implements OnTouchListener {
+
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/Zoom.java]: remove these extra blank lines

File: app/src/main/java/fr/free/nrw/commons/Zoom.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+
+
+    Matrix matrix = new Matrix();
+    Matrix savedMatrix = new Matrix();
+
+
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/Zoom.java]: We usually have one blank line between logical separations.

Please refer to the google style guide. https://google.github.io/styleguide/javaguide.html

File: app/src/main/java/fr/free/nrw/commons/Zoom.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        ImageView view = imageview;
+        view.setOnTouchListener(this);
+    }
+
+    @Override
+    public boolean onTouch(View v, MotionEvent event) {
+        ImageView view = (ImageView) v;
+        view.setScaleType(ImageView.ScaleType.MATRIX);
+        float scale;
+
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/Zoom.java]: same as above

File: app/src/main/java/fr/free/nrw/commons/Zoom.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    private void midPoint(PointF point, MotionEvent event) {
+        float x = event.getX(0) + event.getX(1);
+        float y = event.getY(0) + event.getY(1);
+        point.set(x / 2, y / 2);
+    }
+
+
+}
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/Zoom.java]: add a new line at the end of file

**Comment by tanvidadu** [comment]
@maskaravivek I was asked to make a separate class for implementing zoom feature which will be later used as a part of revamped upload screen

File: app/src/main/java/fr/free/nrw/commons/Zoom.java (hunk 2 above)
Comment (tanvidadu) [reply app/src/main/java/fr/free/nrw/commons/Zoom.java]: will follow the guidelines and make changes accordinly

**Comment by tanvidadu** [comment]
@maskaravivek I am not sure as of where to add this functionality, because no discussion has been there on applying it to the existing code. please help in this respect . Thanks :)

**Comment by nicolas-raoul** [comment]
The upload screen is currently being redesigned, so I would suggest adding just a simple button at the bottom of the upload screen.
Later the redesign will decide what to make with this button.
Thanks!

**Comment by tanvidadu** [comment]
Thanks @nicolas-raoul  I will start working on it :)

**Comment by tanvidadu** [comment]
Made a new pr for this issue and had incorporated all the changes . thanks :)

**Pull request closed without merging**